app/transactions/views.py

```diff
@@ -0,0 +1,39 @@
+from flask import Blueprint, flash, render_template, request
+from flask_login import login_required
+from sqlalchemy.exc import DatabaseError
+
+from app.crud import create_transaction
+from app.transactions.forms import TransactionForm
+
+blueprint = Blueprint(
+    'transactions', __name__, url_prefix='/transactions', template_folder='templates', static_folder='static'
+)
+
+
+# @login_required
+@blueprint.route('/add', methods=['GET', 'POST'])
+def add():
+    page_title = 'Добавление Транзакции'
+    form = TransactionForm()
+    if request.method == 'GET':
+        return render_template('transactions/add.html', page_title=page_title, form=form)
+    if not form.validate():
+        flash('Невалидная форма')
+        return render_template('transactions/add.html', form=form)
+    try:
+        create_transaction(form.data)
+    except DatabaseError:
+        flash('Ошибка создания транзакции')
+    return  # TODO придумать, куда переводить пользователя
```

Comment:
```suggestion
    form = TransactionForm()
    if form.validate_on_submit():
        try:
            create_transaction(form.data)
        except DatabaseError:
            flash('Ошибка создания транзакции')
    else:
        flash('Невалидная форма')
    return render_template('transactions/add.html', page_title=page_title, form=form)
```
Мне кажется хорошо бы придерживаться такой структуры для эндпоинтов. Могу на созвоне словами объяснить почему это хорошо. Но основной тезис - то что так проще всего структура. Один return, и в случае ошибки валидации мы так же возвращаем пользователю уже заполненную форму.

Reply:
А ну, еще в try-else можно добавить редирект в случае успешного создания.